src/chai/ManagedArray.hpp

```diff
@@ -134,6 +134,8 @@ class ManagedArray : public CHAICopyable
    */
   CHAI_HOST_DEVICE ManagedArray(ManagedArray const& other);
 
+  CHAI_HOST_DEVICE ManagedArray(ManagedArray && other);
```

Comment:
What's this for?

Reply:
This along with the move assignment operator perform the default operation, but now also do `other = nullptr`. The reason for clearing `other` is so that if a wrapper class were then to call `other.free()` it wouldn't delete the pointer record now "owned" by `*this`.

Reply:
We didn't have a move operator before though?

Reply:
True, I forgot that. Is there a reason to not have a move constructor?

Reply:
We discussed this before, and decided not to add one (https://github.com/LLNL/CHAI/pull/47) in case it impacts any application correctness I thought.

Reply:
Wow, I can't believe I forgot that. My bad. Is it ok to keep the move assignment operator though? That would be a lot less disruptive. If not it's fine, you can mimic the move semantics with the following
```c++
ManagedArray<T> toMove(10);
ManagedArray<T> x(nullptr);
x = toMove;
toMove = nullptr;
```

Keeping the move assignment operator would condense the last two, so it would be nice but certainly not necessary.

Reply:
I would be fine with move assigment. The rest of this looks good.

Reply:
Done.